conformance/utils/suite/suite.go

```diff
@@ -139,6 +139,8 @@ func (suite *ConformanceTestSuite) Setup(t *testing.T) {
 	t.Logf("Test Setup: Applying programmatic resources")
 	secret := kubernetes.MustCreateSelfSignedCertSecret(t, "gateway-conformance-web-backend", "certificate", []string{"*"})
 	suite.Applier.MustApplyObjectsWithCleanup(t, suite.Client, suite.TimeoutConfig, []client.Object{secret}, suite.Cleanup)
+	secret = kubernetes.MustCreateSelfSignedCertSecret(t, "gateway-conformance-infra", "valid-certificate", []string{"*"})
+	suite.Applier.MustApplyObjectsWithCleanup(t, suite.Client, suite.TimeoutConfig, []client.Object{secret}, suite.Cleanup)
```

Comment:
So with this change we'll have `certificate` and `valid-certificate`. Technically both of them are valid. Should we maybe update the naming here to indicate what's different about these and what they're for?

Perhaps we can even just use the existing `certificate` resource rather than creating a new one?

Reply:
I changed the name to highlight the usage. We need another certificate in `gateway-conformance-infra` to be able to check intra-namespace certificate reference. If we use the certificate in `gateway-conformance-web-backend`, we would also need a `ReferenceGrant`, but these tests are not about it.